src/naive/datetime/serde.rs

```diff
@@ -1020,39 +2422,100 @@ pub mod ts_seconds_option {
+            formatter.write_str("a unix timestamp string in seconds or none")
         }
 
-        /// Deserialize a timestamp in seconds since the epoch
+        /// Deserialize a string timestamp in seconds since the epoch
         fn visit_some<D>(self, d: D) -> Result<Self::Value, D::Error>
         where
             D: de::Deserializer<'de>,
         {
-            d.deserialize_i64(SecondsTimestampVisitor).map(Some)
+            d.deserialize_str(ts_seconds_string::SecondsTimestampVisitor).map(Some)
         }
 
-        /// Deserialize a timestamp in seconds since the epoch
+        /// Deserialize a string timestamp in seconds since the epoch
         fn visit_none<E>(self) -> Result<Self::Value, E>
         where
             E: de::Error,
         {
             Ok(None)
         }
 
-        /// Deserialize a timestamp in seconds since the epoch
+        /// Deserialize a string timestamp in seconds since the epoch
         fn visit_unit<E>(self) -> Result<Self::Value, E>
         where
             E: de::Error,
         {
             Ok(None)
         }
     }
+
+    #[cfg(all(test, feature = "serde"))]
+    mod tests {
+        use crate::naive::serde::ts_seconds_string_option;
+        use crate::{NaiveDate, NaiveDateTime};
+        use serde_derive::{Deserialize, Serialize};
+
+        #[derive(Deserialize, Serialize)]
+        struct S {
+            #[serde(with = "ts_seconds_string_option")]
+            time: Option<NaiveDateTime>,
+        }
+
+        #[test]
+        fn test_serialize_naive_ts_seconds_string_option_some() -> Result<(), serde_json::Error> {
+            let my_s = S {
+                time: Some(
+                    NaiveDate::from_ymd_opt(2015, 5, 15).unwrap().and_hms_opt(10, 0, 0).unwrap(),
+                ),
+            };
+
+            let as_string = serde_json::to_string(&my_s)?;
+
+            assert_eq!(as_string, r#"{"time":"1431684000"}"#);
+
+            Ok(())
+        }
+
+        #[test]
+        fn test_serialize_naive_ts_seconds_string_option_none() -> Result<(), serde_json::Error> {
+            let my_s = S { time: None };
+
+            let as_string = serde_json::to_string(&my_s)?;
+
+            assert_eq!(as_string, r#"{"time":null}"#);
+
+            Ok(())
+        }
+
+        #[test]
+        fn test_deserialize_naive_ts_seconds_string_option_some() -> Result<(), serde_json::Error> {
+            const JSON: &'static str = r#"{"time":"1431684000"}"#;
+
+            let my_s: S = serde_json::from_str(JSON)?;
+
+            assert_eq!(my_s.time.unwrap().timestamp(), 1431684000);
+
+            Ok(())
+        }
+
+        #[test]
+        fn test_deserialize_naive_ts_seconds_string_option_none() -> Result<(), serde_json::Error> {
+            const JSON: &'static str = r#"{"time":null}"#;
+
+            let my_s: S = serde_json::from_str(JSON)?;
+
+            assert_eq!(my_s.time, None);
+
+            Ok(())
+        }
+    }
```

Comment:
Same question about testing maximum and minimum boundaries, and values immediately outside those boundaries.